assets/src/edit-story/components/library/common/searchInput.js

```diff
@@ -70,67 +43,58 @@ const CloseIcon = styled(Close)`
  * @return {SearchInput} The component.
  * @class
  */
-export default function SearchInput({
+function WrappedSearchInput({
   initialValue,
   placeholder,
   onSearch,
-  disabled,
-  incremental,
-  delayMs,
+  disabled = false,
+  incremental = false,
+  delayMs = 500,
 }) {
   // Local state so that we can debounce triggering searches.
   const [localValue, setLocalValue] = useState(initialValue);
 
-  /**
-   * Effectively performs a search, triggered at most every `delayMs`.
-   */
+  // Effectively performs a search, triggered at most every `delayMs`.
   const [changeSearchTermDebounced] = useDebouncedCallback(() => {
     onSearch(localValue);
   }, delayMs);
 
-  /**
-   * Handle search input changes. Triggers with every keystroke.
-   *
-   * @param {string} v the new search term.
-   */
-  const onChange = (v) => {
-    setLocalValue(v);
-    // When in non-incremental mode, we still trigger onSearch when the search
-    // term is empty, so that the user doesn't need to press enter in that case.
-    if (!incremental && v !== '') {
-      return;
-    }
-    if (incremental && delayMs) {
-      changeSearchTermDebounced();
-    } else {
-      onSearch(v);
-    }
-  };
+  const onNewValue = useCallback(
+    (newValue) => {
+      // Handle search input changes. Triggers with every keystroke.
+      setLocalValue(newValue);
+      if (incremental && delayMs) {
+        changeSearchTermDebounced();
+      } else {
+        onSearch(newValue);
+      }
+    },
+    [changeSearchTermDebounced, onSearch, delayMs, incremental]
+  );
+
+  const onChange = useCallback((evt) => onNewValue(evt.target.value), [
+    onNewValue,
+  ]);
 
-  const onKeyDown = (e) => {
-    if (!incremental && e.key === 'Enter') {
-      onSearch(localValue);
-    }
-  };
+  const onClear = useCallback(() => onNewValue(''), [onNewValue]);
+
+  const hasContent = localValue.length > 0;
 
   return (
-    <SearchField>
-      <Search
-        value={localValue}
-        placeholder={placeholder}
-        onChange={onChange}
-        onKeyDown={onKeyDown}
-        disabled={disabled}
-        aria-label={__('Search from library', 'web-stories')}
-        clear
-        clearIcon={<CloseIcon />}
-        showClearIconBackground={false}
-      />
-    </SearchField>
+    <SearchInput
+      inputValue={localValue}
+      placeholder={placeholder}
+      onChange={onChange}
+      handleClearInput={onClear}
+      disabled={disabled}
+      ariaInputLabel={__('Search', 'web-stories')}
```

Comment:
```suggestion
      aria-label={__('Search', 'web-stories')}
```

this is just passed through spreading `...rest` to the input.

Reply:
Thanks, updated.